airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/helpers/CdcConfigurationHelper.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mysql.helpers;
+
+import static java.util.stream.Collectors.toList;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.functional.CheckedConsumer;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import java.sql.SQLException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for MySqlSource used to check cdc configuration in case of:
+ * <p>
+ * 1. adding new source and checking operations #getCheckOperations method.
+ * </p>
+ * <p>
+ * 2. checking whether binlog required from saved cdc offset is available on mysql server
+ * #checkBinlog method
+ * </p>
+ */
+public class CdcConfigurationHelper {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CdcConfigurationHelper.class);
+  private static final String CDC_OFFSET = "mysql_cdc_offset";
+  private static final String LOG_BIN = "log_bin";
+  private static final String BINLOG_FORMAT = "binlog_format";
+  private static final String BINLOG_ROW_IMAGE = "binlog_row_image";
+
+  /**
+   * Method will check whether required binlog is available on mysql server
+   *
+   * @param offset - saved cdc offset with required binlog file
+   * @param database - database
+   */
+  public static void checkBinlog(JsonNode offset, JdbcDatabase database) {
+    Optional<String> binlogOptional = getBinlog(offset);
+    binlogOptional.ifPresent(binlog -> {
+      if (isBinlogAvailable(binlog, database)) {
+        LOGGER.info("""
+                    Binlog %s is available""".formatted(binlog));
+      } else {
+        String error =
+            """
+            Binlog %s is not available. This is a critical error, it means that requested binlog is not present on mysql server. To fix data synchronization you need to reset your data. Please check binlog retention policy configurations."""
+                .formatted(binlog);
+        LOGGER.error(error);
+        throw new RuntimeException("""
+                                   Binlog %s is not available.""".formatted(binlog));
```

Comment:
Should this part of the code executed too if `binlogOptional.ifPresent` is not true?

Reply:
> Should this part of the code executed too if `binlogOptional.ifPresent` is not true?

- binlogOptional.ifPresent check that saved cdc_offset contain information about last sync binlog and returns binlog file name

- if (isBinlogAvailable(binlog, database)) - will check that required for sync binlog file is present on mysql-server

if binlog is not on mysql server anymore only then this part of code will executed.